### PR TITLE
Access cleanup iteration 1

### PR DESCRIPTION
**Pull request by corp-0**
### Purpose
First iteration in my effort to clean up clearance and access chaos.

### Improved Clearance Framework
I tried to make the API clearer so less people try to improvise ways to check clearance. I did it by:
1. Renaming ``IClearanceProvider`` to ``IClearanceSource`` as the interface is supposed to represent a source of clearance definitions, like an ID card for example.
2. Renaming ``ClearanceCheckabl`` to ``ClearanceRestricted`` as adding this component to a game object means you want to **restrict** a functionality behind a clearance check.
3. Adding public methods to ``BasicClearanceSource`` to add and remove individual clearance definitions. Before it could only receive an entire list or be completely cleared.
4. Adding a convenience public method to ``BasicClearanceSource`` that receives a ``IClearanceSource`` and a success and failure Actions. It will check the clearance and do whichever corresponds to the check result. This is intended for the most common case where you just need to check clearance and do something or send an error message back, for example.
5. Starting writing documentation on the dev wiki about the framework and how to do stuff.

### Refactored consumers
I tried moving over as many consumers of the framework trying to change the less code I could. Since this is merely a first iteration, there will be some more work to do, but this is a good start.

### Notes:
Because the way StatusDisplays and Doors is a big pot of spaghetti, I had to undo an exploit prevention @MaxIsJoe did. This means the exploit to open doors you don't have clearance to is there again. I'm going to fix it in the next iteration. 

I already had a refactor done, but the changes were getting way too big. I decided against continuing that effort and start again, in the smallest way possible.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Devices/AccessRemote.cs [hunk 1]
```diff
@@ -83,17 +82,21 @@ public bool WillInteract(HandApply interaction, NetworkSide side)
 			return Validations.HasComponent<DoorMasterController>(interaction.TargetObject) && Validations.CanApply(interaction.PerformerPlayerScript, interaction.TargetObject, side, false, ReachRange.Unlimited);
 		}
 
+		//FIXME: this shouldn't make its own check and use doors like this. Refactor doors to allow interacting with them with devices instead
+		// and let the door handle the interaction.
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Devices/AccessRemote.cs:86]: might be worth adding note to make them client predicted?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Devices/AccessRemote.cs:86]: Uhmm, I was going to do the mentioned refactor. Don't know how to do them client predicted

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Devices/AccessRemote.cs:86]: fair enough just putting a to-domaybe, is fine

File: UnityProject/Assets/Scripts/Systems/Clearance/BasicClearanceSource.cs [hunk 2]
```diff
@@ -76,18 +92,37 @@ public void ServerSetLowPopClearance(IEnumerable<Clearance> newClearance)
...
+		public void ServerRemoveClearance(Clearance forRemoval)
+		{
+			syncedClearance.Remove(forRemoval);
+			netIdentity.isDirty = true;
+		}
+
+		[Server]
+		public void ServerRemoveLowPopClearance(Clearance forRemoval)
+		{
+			syncedLowpopClearance.Remove(forRemoval);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/Clearance/BasicClearanceSource.cs:109]: just saying in most of the SOs low pop has all the entries of normal pop + a few extras

**Comment by Bod9001** [comment]
otherwise looks good

File: UnityProject/Assets/Scripts/Systems/Clearance/BasicClearanceSource.cs (hunk 2 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/Clearance/BasicClearanceSource.cs:109]: That is how it is supposed to be (recreating how it is in byond)

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Systems/Clearance/BasicClearanceSource.cs:109]: Sorry Miss read must've been me being ill

File: UnityProject/Assets/Scripts/Objects/Wallmounts/PublicTerminals/PublicDepartmentTerminal.cs [hunk 3]
```diff
@@ -62,23 +62,24 @@ public enum DepartmentToInt
...
-
-		[HideInInspector]
-		public readonly SyncList<MessageData> receivedMessageData = new SyncList<MessageData>();
+		public GUI_PublicTerminal TerminalGUI { get; set; }
 
-		[HideInInspector]
-		public readonly SyncList<MessageData> archivedMessageData = new SyncList<MessageData>();
+		public MessageData SendMessageData { get; private set; }
+		public SyncList<MessageData> ReceivedMessageData { get; } = new();
+		public SyncList<MessageData> ArchivedMessageData { get; } = new();
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/Wallmounts/PublicTerminals/PublicDepartmentTerminal.cs:66]: just saying with SyncList you have to manually call Set dirty on it

Comment (corp-0) [reply UnityProject/Assets/Scripts/Objects/Wallmounts/PublicTerminals/PublicDepartmentTerminal.cs:66]: this isn't part of my code. Atner did it.

**Review by Bod9001 — DISMISSED** [review]

**Comment by Bod9001** [comment]
reminder do docking on Square station for me todo

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**